### PR TITLE
[#WA]Events statistics

### PR DESCRIPTION
**Pull request by Tanjaa1**

### CODE REVIEW

File: WebApplication/Controller/AppointmentEventController.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        static readonly HttpClient client = new HttpClient();
+
+        [HttpPost("addEvent")]
+        public async Task<IActionResult> GetAllAppointmentsByPatientId(PatientAppointmentEventParams patientAppointmentEventParams)
+        {
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(Request.Headers["Authorization"].ToString().Split(" ")[0]
+                                                                                                    , Request.Headers["Authorization"].ToString().Split(" ")[1]);
+            var parameter = new StringContent(JsonConvert.SerializeObject(patientAppointmentEventParams, Formatting.Indented), Encoding.UTF8, "application/json");
+
+            HttpResponseMessage response = await client.PostAsync("http://localhost:57056/appointmentEvent/addEvent/", parameter);
```
Comment (milicasiriski) [inline WebApplication/Controller/AppointmentEventController.cs]: The url must be created from environment variables (just like in other controllers)

File: WebApplication/Controller/AppointmentEventController.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            response.EnsureSuccessStatusCode();
+            return JsonConvert.DeserializeObject<IActionResult>(await response.Content.ReadAsStringAsync());
+        }
+
+        [HttpGet("eventStatistic")]
+        public async Task<EventStatisticDTO> GetAllEvents()
+        {
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(Request.Headers["Authorization"].ToString().Split(" ")[0]
+                                                                                                     , Request.Headers["Authorization"].ToString().Split(" ")[1]);
+            HttpResponseMessage response = await client.GetAsync("http://localhost:57056/appointmentEvent/all");
```
Comment (milicasiriski) [inline WebApplication/Controller/AppointmentEventController.cs]: Same here

File: project-generated-code-backend/Backend/Controller/SuperintendentControllers/SuperintendentMedicineController.cs [hunk 3]
```diff
@@ -2,7 +2,8 @@
 using HealthClinicBackend.Backend.Service.MedicineService;
 using System.Collections.Generic;
 
-namespace HealthClinicBackend.Backend.Controller.SuperintendentControllers
+namespace HealthClinicBackend.Backend.Controller.
+    intendentControllers
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Controller/SuperintendentControllers/SuperintendentMedicineController.cs]: Why is this changed?

File: project-generated-code-backend/Backend/Events/PatientRegisteredEventLogging/LogPatientAppointmentEventService.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        }
+
+        public void LogEvent(PatientAppointmentEventParams eventParams)
+        {
+            //var patientRegisteredEvent = new PatientRegisteredEvent
+            //{ TimeStamp = DateTime.Now, PatientAge = eventParams.PatientAge };
+
+            //_patientAppointmentEventRepository.LogEvent(PatientAppointmentEvent);
+            throw new NotImplementedException();
+        }
```
Comment (milicasiriski) [inline project-generated-code-backend/Backend/Events/PatientRegisteredEventLogging/LogPatientAppointmentEventService.cs]: Either delete the comments or implement the method

File: project-generated-code-backend/Migrations/20210118114610_migration1.Designer.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
+﻿// <auto-generated />
```
Comment (milicasiriski) [inline project-generated-code-backend/Migrations/20210118114610_migration1.Designer.cs]: Delete migration files and folder

**Review by milicasiriski — CHANGES_REQUESTED** [review]

File: project-generated-code-backend/Backend/Controller/SuperintendentControllers/SuperintendentMedicineController.cs (hunk 3 above)
Comment (maka998) [reply project-generated-code-backend/Backend/Controller/SuperintendentControllers/SuperintendentMedicineController.cs]: By mistake :)

File: MicroServiceAppointment/Backend/Service/AppointmentService.cs [hunk 6]
```diff
@@ -39,26 +42,33 @@ public class AppointmentService : IAppointmentService
...
+        public AppointmentService(ISpecializationRepository specializationRepository,
+                IAppointmentRepository appointmentRepository, IPhysicianRepository physicianRepository,
+                IPatientRepository patientRepository, IAddressRepository addressRepository)
+        {
+            _appointmentRepository = appointmentRepository;
+            _physicianRepository = physicianRepository;
+            _specializationRepository = specializationRepository;
+            _addressRepository = addressRepository;
+            _patientRepository = patientRepository;
+        }
```
Comment (milicasiriski) [inline MicroServiceAppointment/Backend/Service/AppointmentService.cs]: I think you can only have one constructor if you're using dependency injection so delete the one with less parameters

**Review by milicasiriski — CHANGES_REQUESTED** [review]
That's it for the urls, just check that service with multiple constructors as it could cause bugs

**Review by milicasiriski — APPROVED** [review]

**Review by VladimirBudjen — APPROVED** [review]
